internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -7,26 +7,137 @@ import (
 )
 
 var _ = Describe("NGINX Runtime Manager", func() {
 	It("returns whether or not we're using NGINX Plus", func() {
-		mgr := NewManagerImpl(nil, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr := runtime.NewManagerImpl(nil, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeFalse())
 
-		mgr = NewManagerImpl(&ngxclient.NginxClient{}, nil, logr.New(GinkgoLogr.GetSink()))
+		mgr = runtime.NewManagerImpl(&ngxclient.NginxClient{}, nil, zap.New(), nil, nil)
 		Expect(mgr.IsPlus()).To(BeTrue())
 	})
+
+	var (
+		err             error
+		manager         runtime.Manager
+		upstreamServers []ngxclient.UpstreamServer
+		ngxPlusClient   *runtimefakes.FakeNginxPlusClient
+		process         *runtimefakes.FakeProcessHandler
+
+		metrics      *runtimefakes.FakeMetricsCollector
+		verifyClient *runtimefakes.FakeVerifyClient
+	)
+
+	BeforeEach(func() {
+		upstreamServers = []ngxclient.UpstreamServer{
+			{},
+		}
+	})
+
+	Context("Reload", func() {
+		BeforeEach(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			process = &runtimefakes.FakeProcessHandler{}
+			metrics = &runtimefakes.FakeMetricsCollector{}
+			verifyClient = &runtimefakes.FakeVerifyClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+		})
+
+		It("NGINX configuration reload is successful", func() {
+			Expect(manager.Reload(context.Background(), 1)).To(Succeed())
+
+			Expect(process.FindMainProcessCallCount()).To(Equal(1))
+			Expect(process.ReadFileCallCount()).To(Equal(1))
+			Expect(process.KillCallCount()).To(Equal(1))
+			Expect(metrics.IncReloadCountCallCount()).To(Equal(1))
+			Expect(verifyClient.WaitForCorrectVersionCallCount()).To(Equal(1))
+			Expect(metrics.ObserveLastReloadTimeCallCount()).To(Equal(1))
+		})
+
+		When("NGINX configuration reload is not successful", func() {
+			It("should panic if MetricsCollector not enabled", func() {
+				metrics = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					err = manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("should panic if VerifyClient not enabled", func() {
+				metrics = &runtimefakes.FakeMetricsCollector{}
+				verifyClient = nil
+				manager = runtime.NewManagerImpl(ngxPlusClient, metrics, zap.New(), process, verifyClient)
+
+				reload := func() {
+					err = manager.Reload(context.Background(), 0)
+				}
+
+				Expect(reload).To(Panic())
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+	})
+
+	When("running NGINX plus", func() {
+		BeforeEach(func() {
+			ngxPlusClient = &runtimefakes.FakeNginxPlusClient{}
+			manager = runtime.NewManagerImpl(ngxPlusClient, nil, zap.New(), nil, nil)
+		})
+
+		It("successfully updates HTTP server upstream", func() {
+			Expect(manager.UpdateHTTPServers("test", upstreamServers)).To(Succeed())
```

Comment:
Let's also check that we called the mock's `UpdateHTTPServers` method here with the input "test"